tremor-script/src/ast/deploy/raw.rs

```diff
@@ -0,0 +1,785 @@
+        }
+
+        // We iterate over the deployment parameters and filter/ban unknown arguments
+        for name in spec.iter() {
+            let cow_name = Cow::owned(name.to_string());
+            if !args_as_params.contains_key(&cow_name) {
+                // FIXME Add extra args hygienic error
+                return Err(ErrorKind::DeployRequiredArgDoesNotResolve(
+                    outer.extent(&helper.meta),
+                    outer.extent(&helper.meta),
+                    name.to_string(),
+                )
+                .into());
+            }
+        }
+    }
+
+    Ok(args)
+}
+
+impl<'script> Upable<'script> for PipelineDeclRaw<'script> {
+    type Target = PipelineDecl<'script>;
+    fn up<'registry>(self, helper: &mut Helper<'script, 'registry>) -> Result<Self::Target> {
+        // NOTE As we can have module aliases and/or nested modules within script definitions
+        // that are private to or inline with the script - multiple script definitions in the
+        // same module scope can share the same relative function/const module paths.
+        //
+        // We add the script name to the scope as a means to distinguish these orthogonal
+        // definitions. This is achieved with the push/pop pointcut around the up() call
+        // below. The actual function registration occurs in the up() call in the usual way.
+        //
+        helper.module.push(self.id.clone());
+        let (spec, args) = arg_spec_resolver(self.args, &self.params, helper)?;
+        let query = self.query.clone().up_script(helper)?;
+
+        let query_decl = PipelineDecl {
+            mid: helper.add_meta_w_name(self.start, self.end, &self.id),
+            module: helper.module.clone(),
+            id: self.id,
+            spec: spec.clone(),
+            args,
+            query_raw: self.query.clone(),
+            query, // The runnable is redefined by `deploy` statements which complete/finalize and seal the runtime arguments
+        };
+        helper.module.pop();
+        let script_name = query_decl.fqsn(&helper.module);
+        helper
+            .pipeline_defns
+            .insert(script_name, query_decl.clone());
+        Ok(query_decl)
+    }
+}
+
+pub type DeployStmtsRaw<'script> = Vec<DeployStmtRaw<'script>>;
+
+/// we're forced to make this pub because of lalrpop
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct ConnectorDeclRaw<'script> {
+    pub(crate) start: Location,
+    pub(crate) end: Location,
+    pub(crate) id: String,
+    pub(crate) kind: (Vec<IdentRaw<'script>>, IdentRaw<'script>),
+    pub(crate) args: WithArgsRaw<'script>,
+    pub(crate) params: Option<DeployWithExprsRaw<'script>>,
+}
+
+impl<'script> Upable<'script> for ConnectorDeclRaw<'script> {
+    type Target = ConnectorDecl<'script>;
+    fn up<'registry>(self, helper: &mut Helper<'script, 'registry>) -> Result<Self::Target> {
+        // NOTE As we can have module aliases and/or nested modules within script definitions
```

Comment:
<3 nice solution I love it :D

Reply:
Lol. I am not a fan of this solution at all - but its simple and it works! :D